tools/swift-format/swift-format.cpp

```diff
@@ -0,0 +1,344 @@
+
+    static cl::opt<bool> OutputXML("output-replacements-xml",
+                                   cl::desc("Output replacements as XML."),
+                                   cl::cat(SwiftFormatCategory));
+
+    static cl::list<std::string> FileNames(cl::Positional, cl::desc("[<file> ...]"),
+                                           cl::cat(SwiftFormatCategory));
+
+    CodeFormatOptions FormatOptions;
+
+    class FormatterDocument {
+      SourceManager SM;
+      unsigned BufferID;
+      CompilerInvocation CompInv;
+      std::unique_ptr<ParserUnit> Parser;
+
+      class FormatterDiagConsumer: public swift::DiagnosticConsumer {
+
+        void handleDiagnostic(SourceManager &SM, SourceLoc Loc,
+                              DiagnosticKind Kind, StringRef Text,
+                              const DiagnosticInfo &Info) override {
+          llvm::errs() << "Parse error: " << Text << "\n";
+        }
+
+      } DiagConsumer;
+
+    public:
+      FormatterDocument(std::unique_ptr<llvm::MemoryBuffer> Buffer) {
+        updateCode(std::move(Buffer));
+      }
+
+      void updateCode( std::unique_ptr<llvm::MemoryBuffer> buff ) {
+        BufferID = SM.addNewSourceBuffer(std::move(buff));
+
+        Parser.reset(
+                     new ParserUnit(SM, BufferID,
+                                    CompInv.getLangOptions(),
+                                    CompInv.getModuleName())
+                     );
+
+        Parser->getDiagnosticEngine().addConsumer(DiagConsumer);
+        auto &P = Parser->getParser();
+
+        bool Done = false;
+        while (!Done) {
+          P.parseTopLevel();
+          Done = P.Tok.is(tok::eof);
+        }
+      }
+
+      std::pair<LineRange, std::string> reformat(LineRange Range, CodeFormatOptions Options) {
+          return swift::ide::reformat(Range, FormatOptions, SM, Parser->getSourceFile());
+      }
+
+      const MemoryBuffer &memBuffer() const {
+        return *SM.getLLVMSourceMgr().getMemoryBuffer(BufferID);
+      }
+
+      std::pair<unsigned, unsigned>
+      getLineAndColumn(unsigned ByteOffset) const {
+        auto &Buf = memBuffer();
+        if (ByteOffset > Buf.getBufferSize())
+          return std::make_pair(0, 0);
+
+        SMLoc Loc = SMLoc::getFromPointer(Buf.getBufferStart() + ByteOffset);
+        return SM.getLLVMSourceMgr().getLineAndColumn(Loc, BufferID);
+      }
+
+    };
+
+    // Parses <start line>:<end line> input to a pair of line numbers.
+    // Returns true on error.
+    static bool parseLineRange(StringRef Input, unsigned &FromLine,
+                               unsigned &ToLine) {
+      std::pair<StringRef, StringRef> LineRange = Input.split(':');
+      return LineRange.first.getAsInteger(0, FromLine) || LineRange.second.getAsInteger(0, ToLine);
+    }
+
+    static void outputReplacementXML(StringRef Text) {
+      // FIXME: When we sort includes, we need to make sure the stream is correct
```

Comment:
Left-overs from clang-format?  swift doesnt have includes ... I dont think that imports get sorted though.